phpdotnet/phd/Package/PHP/XHTML.php

```diff
@@ -555,37 +622,55 @@ public function format_type_text($type, $tagname) {
         case "double":
             $href = "language.types.float";
             break;
+        case "?":
+            $href = "language.types.null";
+            break;
+        case "false":
+        case "true":
+            $href = "language.types.value";
+            break;
+        case "null":
+            if ($this->simple_nullable) {
+                return "";
+            }
         case "boolean":
         case "integer":
```

Comment:
Those, plus the `"double"` case should warn IMHO, as those are an alternative spelling for the proper built-in types that should not exist in the documentation any more. But can be a follow-up PR.

Reply:
I was thinking about this and other warnings. There's a whole range of formatting and logic errors we could catch in Phd from whitespace in all sorts of tags, incompatible type combinations, etc.
If this was implemented as some sort of option that Phd could be called from the command line with, this whole doc checker could be added as an additional step to the documentations' pull request workflow. Maybe?

Reply:
I think this makes sense, but I was certain we already had support for warnings in PhD as I had seen them :thinking: 

Maybe if it's a new render that runs fast and doesn't need indexing, having it run on CI would make sense. Last time it came up there were concerns about the CI taking too long, or running too frequently as a PR might be update often in quick successions

Reply:
Oh, yeah. The error handler is right there in `functions.php`, a file included in everything Phd which I also never looked inside. :-)

So this format would be called with the `--noindex` option, it would have to override the `createLink()`, `update()` and `appendData()` methods at a minimum, and it's validation method(s) could return `null` and would raise warnings on validation errors. Sounds fairly simple, so it's probably far from it. :-)

Reply:
As everything with PhD :D it looks simple and it isn't :D